src/main/scripts/src/app/shared/perspectives/documents/post-it/attribute-tree/attribute-tree.component.ts

```diff
@@ -28,17 +31,151 @@ import {Component, Input} from '@angular/core';
 export class AttributeTreeComponent {
 
   @Input()
-  public children: object;
+  public index: number;
 
   @Input()
-  public root: boolean;
+  public attributes: Attribute[];
 
-  public isArray = element => Array.isArray(element);
+  @Input()
+  public set data(attributes: object) {
+    for (let key of Object.keys(attributes)) {
+
+      let value;
+      if (typeof attributes[key] === 'string') {
+        value = attributes[key];
+      } else {
+        value = JSON.stringify(attributes[key], undefined, 2);
+      }
+
+      this.attributeInputs.push({
+        name: key,
+        value: value,
+        previousName: value
+      });
+    }
+  }
+
+  @Input()
+  public editable: boolean;
+
+  @Output()
+  public attributeChangeEvent = new EventEmitter<DocumentAttribute>();
+
+  public attributeInputs = [] as DocumentAttribute[];
+
+  public newAttributeInput = {} as DocumentAttribute;
+
+  private selectedInput = {} as AttributeInputElement;
+
+  public generateId(x: number, y: number) {
+    return `Input${this.index}[${x}, ${y}]`;
+  }
+
+  public possibleAttributes(): string[] {
+    return this.attributes
+      .sort((a, b) => a.count > b.count ? -1 : a.count < b.count ? 1 : 0) // sort by frequency
+      .map(attribute => attribute.name);
+  }
+
+  public onInputKey(event: KeyboardEvent): void {
+    this.checkSelectedInputChange(event);
+  }
+
+  private checkSelectedInputChange(event: KeyboardEvent) {
+    switch(event.key) {
```

Comment:
I personally do not like switch statements, what do you think about this?
```typescript
private const eventMapper = {
  ArrowUp: () => this.selectAdjacentInput(0, -1, true),
  ArrowDown: () => this.selectAdjacentInput(0, 1, true),
  ArrowLeft: () => this.selectAdjacentInput(-1, 0, true),
  ArrowRight: () => this.selectAdjacentInput(1, 0, true),
  Enter: () => this.selectedInput.x === 1 ? this.selectAdjacentInput(-1, 1) : this.selectAdjacentInput(1, 0)
}
```
And call it like:
```typescript
eventMapper.hasOwnProperty(event.key) && eventMapper[event.key];
```